Assets/Scripts/WinOrLoseScript.cs

```diff
@@ -54,88 +164,266 @@ void Start()
+        }
+
+        int index = losers.Count - 1; //starts on player in last
+
+        //Generate list of last place players
+        List<int> podiumsToDestroy = new List<int>();
+        podiumsToDestroy.Add(index); //add known biggest loser to list
+
+        while (index - 1 != -1 && IsTied(losers[index], losers[index - 1]))
+        {//while in bounds and has the same score as last place person
+            //add to the destroy list
+            podiumsToDestroy.Add(index - 1);
+
+            //setup for next loop
+            index--;
+        }
+
+        return podiumsToDestroy;
+    }
+
+    void Order()
+    {
+        for(int current = 0; current < players.Length; current++)
+        {
+            for(int after = current + 1; after < players.Length; after++)
             {
-                if(scores[after].score > scores[current].score)
+                if(players[after].score > players[current].score)
                 {
-                    //swap scores
-                    Player temp = scores[current];
-                    scores[current] = scores[after];
-                    scores[after] = temp;
+                    //swap players
+                    Player temp = players[current];
+                    players[current] = players[after];
+                    players[after] = temp;
                 }
             }
         }
+    }
+    #endregion
 
-        return scores;
+    #region Win or Lose
+    void PlayWinOrLose(Player player)
+    {
+        if (winners.Contains(player))
+        {//if local player won
+            Win(player);
+            return;
+        }
+        
+        //otherwise if you didnt win, oyu lost
+        Lose();
     }
-    
-    void Win()
+
+    void Win(Player winner)
     {
-        audioSource.PlayOneShot(winSound);
+        PlaySound(audioVaraibles.winSound);
         text.text = "YOU WON!";
+        winner.GetComponent<Animator>().SetBool("First", true); //plays the winning animation now because the 1-4 animations are already playing
     }
 
     void Lose()
     {
-        audioSource.PlayOneShot(loseSound);
+        PlaySound(audioVaraibles.loseSound);
         text.text = "You Lost";
     }
+    #endregion
+
+    #region helper functions
+    Player localPlayerCache = null;
+    bool IsLocalPlayer(Player player)
+    {
+        if (localPlayerCache == null)
+        {//if not cached
+            bool result = player.GetComponent<PhotonView>().Controller.IsLocal;
+            if (result)
+            {
+                localPlayerCache = player;
+                return true;
+            }
+        } else
+        {
+            if(player == localPlayerCache)
+            {
+                return true;
+            }
+        }
+        
+        return false;
+    }
+
+    void PlaySound(AudioClip audioClip)
+    {
+        audioVaraibles.audioSource.PlayOneShot(audioClip);
+    }
+    #endregion
 
     #region Button Functions
```

Comment:
we should really just make a button manager script and put all the button code in there. Also scene loading will not work that way.

Reply:
Agreed. I will get to that on a later date.